### PR TITLE
Explicit color support for range of n 

### LINKED ISSUE #28: Question about color support.
For Qualitative schemes, restrict number of colors to 6?

Is the any solution to support more color? as the Paired has 12 kinds of colors.

### PR DESCRIPTION
**Pull request by nanodan**
Moved the hex color values to external files for easier management and hard-coded all ranges that exist on colorbrewer so less interpolation is necessary: will now match up exactly with the website if it exists at that length, and otherwise will interpolate. Allows for variety of lengths for qualitative now as well.

(closes #28)

### CODE REVIEW

**Comment by nanodan** [comment]
Okay this is good to go - review and let me know!

File: branca/colormap.py [hunk 1]
```diff
@@ -9,223 +9,25 @@
 from __future__ import absolute_import
 
 import math
+import json
+import pkg_resources
```
Comment (ocefpaf) [inline branca/colormap.py]: Not sure if we should be using `pkg_resources`, there are other ways...

File: branca/colormap.py [hunk 2]
```diff
@@ -9,223 +9,25 @@
...
-            'Pastel1': ['#fbb4ae', '#b3cde3', '#ccebc5',
-                        '#decbe4', '#fed9a6', '#ffffcc'],
-            'Pastel2': ['#b3e2cd', '#fdcdac', '#cbd5e8',
-                        '#f4cae4', '#e6f5c9', '#fff2ae'],
-            'Set1': ['#e41a1c', '#377eb8', '#4daf4a',
-                     '#984ea3', '#ff7f00', '#ffff33'],
-            'Set2': ['#66c2a5', '#fc8d62', '#8da0cb',
-                     '#e78ac3', '#a6d854', '#ffd92f'],
-            'Set3': ['#8dd3c7', '#ffffb3', '#bebada',
-                     '#fb8072', '#80b1d3', '#fdb462']}
```
Comment (ocefpaf) [inline branca/colormap.py]: I don't have a strong opinion on if these should be in the file or as an external resource. I'll leave that to @BibMartin to decide.

File: branca/colormap.py [hunk 3]
```diff
@@ -9,223 +9,25 @@
...
+resource_path_schemes = '/_schemes.json'
+resource_path_cnames = '/_cnames.json'
+
+cnames_string = pkg_resources.resource_stream(
+    resource_package, resource_path_cnames).read().decode()
+_cnames = json.loads(cnames_string)
+
+schemes_string = pkg_resources.resource_stream(
+    resource_package, resource_path_schemes).read().decode()
+_schemes = json.loads(schemes_string)
```
Comment (ocefpaf) [inline branca/colormap.py]: We can probably ditch package `pkg_resources` and use the `os` module like:

```python
rootpath = os.path.abspath(os.path.dirname(__file__))
resource_path_schemes = os.path.join(rootpath, '_schemes.json')
```

**Comment by ocefpaf** [comment]
@nanodan note that @BibMartin is quite busy with the day job, so if he cannot review this in the next few days please ping me again and I'll merge this.

**Comment by nanodan** [comment]
@ocefpaf I'll check in on the weekend and ping you if still outstanding.

File: branca/colormap.py (hunk 3 above)
Comment (nanodan) [reply branca/colormap.py]: I'll leave it up to you, just let me know. I can re-write for os.path; I just thought pkg_resources was cleaner. I'm not sure either way though if one approach or the other is best practice.

**Comment by nanodan** [comment]
@ocefpaf did you want to merge this?

**Pull request merged**